src/daemon/keymap.c

```diff
@@ -682,7 +682,7 @@ void process_input_urb(void* context, unsigned char* buffer, int urblen, ushort
                     if(urblen == 64) {
                         corsair_kbcopy(targetkb->input.keys, buffer + 2);
                     } else if(firstbyte == NKRO_KEY_IN || firstbyte == NKRO_MEDIA_IN) {
-                        if(!targetkb->active)
+                        if(!targetkb->active || (firstbyte == NKRO_MEDIA_IN && BRAGI_HAS_MEDIA_MACRO(targetkb)))
```

Comment:
In my case at least this is not going to be called as the Wireshark displays "Bragi extra key input" 
and the "Extra HID" is as below:
```
0000   00 2c c9 85 b0 99 ff ff 43 01 83 04 08 00 2d 00   .,......C.....-.
0010   7c 19 29 62 00 00 00 00 3c f9 08 00 00 00 00 00   |.)b....<.......
0020   40 00 00 00 40 00 00 00 00 00 00 00 00 00 00 00   @...@...........
0030   01 00 00 00 00 00 00 00 00 02 00 00 00 00 00 00   ................
0040   00 02 00 00 00 00 00 00 00 00 02 00 00 00 00 00   ................
0050   00 04 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0060   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0070   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................

```

That will satisfy both the `kb->protocol == PROTO_BRAGI` and the `urblen == 64` so it will call the `corsair_kbcopy` function.

Reply:
urblen is not 64 in hardware media key packets. If I am not mistaken, they are either 3 or 4 bytes.
Please try capturing volume up without ckb-next running.

I am starting to suspect though that for some reason hardware mode media key packets aren't sent in software mode, which would explain why this doesn't work. For clarification the extra key input packets are the software packets, and the HID input ones are the hardware ones.